### PR TITLE
Remove all Bazel artifacts

### PR DESCRIPTION
**Pull request by tylerferrara**
#### What type of PR is this?
/kind cleanup
/kind deprecation

#### What this PR does / why we need it:
Since Bazel is no longer relied upon within CIP, this change deletes all Bazel-related files. Additionally, all documentation reflects this complete deprecation, removing all mention of Bazel. 

Part of the effort to [Reduce Build Maintenance](https://docs.google.com/document/d/1tQTb8dqKQtsL1a5jmCDbRg2MEDMkAnT07vrrxyZxLfk/edit?usp=sharing).
#### Which issue(s) this PR fixes:
#300 

#### Does this PR introduce a user-facing change?

```release-note
Remove Bazel-related files from the repository.
Update Docs to reflect Bazel deprecation.
```
cc: @listx @amwat @justaugustus @kubernetes-sigs/release-engineering

### CODE REVIEW

**Comment by tylerferrara** [comment]
NOTE: `dependencies.yaml` must be kept, as it's used in the [verify-dependencies](https://github.com/kubernetes-sigs/k8s-container-image-promoter/blob/master/Makefile#L111) make target, currently triggered by the pre-submit: `pull-cip-verify`. However, if this is no longer needed, please indicate as such, and I can make an accompanying PR.

**Comment by justaugustus** [comment]
@tylerferrara -- Please leave `dependencies.yaml` in place as it's used to track non-bazel dependencies.

**Comment by tylerferrara** [comment]
Additionally, should `legacy/` remain untouched? This function is being referenced in many other legacy programs:
```golang
func bazelTestPath(testName string, paths ...string) string {
	prefix := []string{
		os.Getenv("PWD"),
		"inventory_test",
		testName,
	}
	return filepath.Join(append(prefix, paths...)...)
}
```

**Comment by listx** [comment]
> Additionally, should `legacy/` remain untouched? This function is being referenced in many other legacy programs:
> 
> ```go
> // legacy/dockerregistry/inventory_test.go
> func bazelTestPath(testName string, paths ...string) string {
> 	prefix := []string{
> 		os.Getenv("PWD"),
> 		"inventory_test",
> 		testName,
> 	}
> 	return filepath.Join(append(prefix, paths...)...)
> }
> ```
> 
> Even just changing the name may help to reduce confusion.
> What do you think? @listx @justaugustus @amwat

Yeah we should change the name. You should do this in a separate commit (but within this PR), because renames (that don't change any behavior) are special and should be isolated from other changes.

File: VERSION [hunk 1]
```diff
@@ -1 +0,0 @@
-1.337.0
```
Comment (listx) [inline VERSION:1]: Is this file necessarily tied to Bazel?

**Review by justaugustus — APPROVED** [review]

File: hack/verify-archives.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 # About:
 # This script extracts the golden image tarballs and compares their contents against the
-# already-tracked golden folder (which is used by Bazel to generate deterministic images
+# already-tracked golden folder (which is used by Docker to generate deterministic images
```
Comment (listx) [inline hack/verify-archives.sh:19]: Docker does not generate deterministic images.

That said, this parenthetical is worth deleting entirely.

File: VERSION (hunk 1 above)
Comment (justaugustus) [reply VERSION:1]: Good catch. Let's restore this.

File: Makefile [hunk 3]
```diff
@@ -22,7 +22,7 @@ all: test
 ##@ Build
 .PHONY: build
 build: ## Go build
-	go build -o cip ${REPO_ROOT}/cmd/cip/...
+	${REPO_ROOT}/go_with_version.sh build ${REPO_ROOT}/cmd/cip
```
Comment (tylerferrara) [inline Makefile:25]: This should have been merged with #324 
When building, we currently have an incomplete version stamping.
```bash
$ cip version
GitVersion:    
GitCommit:     
GitTreeState:  
BuildDate:     
GoVersion:     go1.16.5
Compiler:      gc
Platform:      linux/amd64
```

With this change, we get all the information we want.
```bash
$ cip version
GitVersion:    v3.0.0
GitCommit:     999f2538d2d4a63a1b5c389d7b53411e9ad24226
GitTreeState:  clean
BuildDate:     2021-07-07T21:06:08Z
GoVersion:     go1.16.5
Compiler:      gc
Platform:      linux/amd64
```

File: cmd/cip/cmd/root.go [hunk 4]
```diff
@@ -26,17 +26,6 @@ import (
 	"sigs.k8s.io/release-utils/log"
 )
 
-var (
```
Comment (tylerferrara) [inline cmd/cip/cmd/root.go:29]: This is part of the stamping fix, as mentioned above.

File: hack/verify-archives.sh (hunk 2 above)
Comment (tylerferrara) [reply hack/verify-archives.sh]: ack.

**Review by justaugustus — APPROVED** [review]
@tylerferrara -- Great work!
@listx -- the review comments look to all be addressed, but if there's anything outstanding, let's take that as a follow-up.

/hold cancel

**Pull request merged**